Added examples for proto yab in Readme (#308)
yarpc/yab: README.md

yab -t ~/keyvalue.thrift -p localhost:12345 keyvalue KeyValue::get -r '{"key": "hello"}'
```

#### Proto
The following examples assume that the Proto service running looks like:
```proto
service KeyValue {
  rpc GetValue(Request) returns (Response) {}  

> }

Reviewer: You should also add the definition of `Request` and `Response` since there are not a well known type (this differs from Thrift since you can not use string here)

yab -t ~/keyvalue.thrift -p localhost:12345 keyvalue KeyValue::get -r '{"key": "hello"}'
```

#### Proto
The following examples assume that the Proto service running looks like:
```proto
service KeyValue {
  rpc GetValue(Request) returns (Response) {}  
}
```

If a gRPC service was running with name `KeyValue` on `localhost:12345` with proto package name `pkg.keyvalue` and 
binary file containing a compiled protobuf FileDescriptorSet as `keyValue.proto.bin`, you can
make a call to the `GetValue` method by running:

```bash
yab keyvalue pkg.keyvalue/GetValue --file-descriptor-set-bin=keyValue.proto.bin -r '{"key": "hello"} -p localhost:12345'
```
Protobuf FileDescriptorSet can be generated by running:
```bash
protoc --include_imports --descriptor_set_out=keyValue.proto.bin keyValue.proto
```

> Note : if Server Reflection is enabled which provides information about publicly-accessible gRPC services on a server, then
> no need to specify the FileDescriptorSet binary.

Reviewer 2: ```suggestion
Note : If [Server Reflection](https://github.com/grpc/grpc/blob/master/doc/server-reflection.md) is enabled which provides information about publicly-accessible gRPC services on a server, then there is no need to specify the FileDescriptorSet binary:

```bash
yab keyvalue pkg.keyvalue/GetValue -r '{"key": "hello"} -p localhost:12345'
```

#### Specifying Peers

> This specifies a single `host:port` using `-p`, but you can also specify multiple peers

Reviewer 2: ```suggestion
A single `host:port` is specified using `-p`, but you can also specify multiple peers
```

by passing the `-p` flag multiple times:
```bash
yab -t ~/keyvalue.thrift -p localhost:12345 -p localhost:12346 keyvalue KeyValue::get -r '{"key": "hello"}'
```
```bash
yab keyvalue pkg.keyvalue/GetValue -r '{"key": "hello"} -p localhost:12345 -p localhost:12346'
```

If you have a file containing a list of host:ports (either JSON or new line separated), you can
specify the file using `-P`:
```bash
yab -t ~/keyvalue.thrift -P ~/hosts.json keyvalue KeyValue::get -r '{"key": "hello"}'
```
```bash

> yab keyvalue pkg.keyvalue/GetValue -r '{"key": "hello"} -p ~/hosts.json'

Reviewer 2: ```suggestion
yab keyvalue pkg.keyvalue/GetValue -r '{"key": "hello"} -P ~/hosts.json'
```

should be capital P for peer list

```

`yab` also supports HTTP, instead of the peer being a single `host:port`, you would use a URL:
```bash
yab -t ~/keyvalue.thrift -p "http://localhost:8080/rpc" keyvalue KeyValue::get -r '{"key": "hello"}'
```

> 

Reviewer 2: suggest leaving the blank line between sections

```bash
yab -t ~/keyvalue.thrift -p "http://localhost:8080/rpc" keyvalue KeyValue::get -r '{"key": "hello"}'
```
#### Proto
The following examples assume that the Proto service running looks like:
```proto
service KeyValue {
  rpc GetValue(Request) returns (Response) {}  
}
```

> If a gRPC service was running with name `KeyValue` on `localhost:12345` with proto package name `pkg.keyvalue` and 

Reviewer 2: we also should mention that if reflection is enabled on the server, there's no need to specify the binary file

Author: Good catch! added a note.

> binary file containing a compiled protobuf FileDescriptorSet as `keyValue.proto.bin`, you can

Reviewer 2: Can you add a short note on how to create the `.proto.bin` file using the `protoc` compiler as well?

Author: Done

make a call to the `get` method by running:

```bash
yab keyvalue pkg.keyvalue/GetValue --file-descriptor-set-bin=keyValue.proto.bin -r '{"key": "hello"} -p localhost:12345'
```

This specifies a single `host:port` using `-p`, but you can also specify multiple peers
by passing the `-p` flag multiple times:
```bash
yab keyvalue pkg.keyvalue/GetValue --file-descriptor-set-bin=keyValue.proto.bin -r '{"key": "hello"} -p localhost:12345 -p localhost:12346'
```

If you have a file containing a list of host:ports (either JSON or new line separated), you can
specify the file using `-P`:
```bash
yab keyvalue pkg.keyvalue/GetValue --file-descriptor-set-bin=keyValue.proto.bin -r '{"key": "hello"} -p ~/hosts.json'

> ```

Reviewer 2: rather than duplicating the port section, maybe we should add a new sub-section for specifying peers and avoid the duplication under thrift / proto?

Author: Separated out section for this :)
